### PR TITLE
Fixes diacritics regression #7484 #7450 #7560

### LINKED ISSUE #7450: Accented characters in movies and people's name should be treated as non-accented
Take a Director, some metadata provider name him `Kieślowski` some others will name him `Kieslowski`

As of now (I'm using 10.8.0.-a5), these are two different directors.
He will appear as two different person, and related movies will be "director unrelated".

I guess that should be a very generic behaviours for all string metadata.

Moreover, I realize "original name" metadata for people (like a person's name in its own language script) are unsearchable.

### PR DESCRIPTION
**Pull request by Jellifi007**
**Changes**

Fixes diacritics regression introduced by Bond-009@981cf4cfa0d45a3654578a1989738fc82469c889

Example: searching "jon" will now correctly return movies containing "Jönsson":
image

**Issues**

Fixes #7450 #7484 #7560

### CODE REVIEW

**Comment by Jellifi007** [comment]
Also note that:

- .NET 6.0 seems to replace automatically each invalid UTF-16 code point (ill-formed surrogates for example) by code point \uFFFD (replacement char).
- Xunit.InlineDataAttribute seems to replace each invalid UTF-16 code point by two \uFFFD chars (not sure why)

So the "_nonConformingUnicode" regex has to be slightly modified to take this in account and strip those unecessary \uFFFD chars.

**Comment by crobibero** [comment]
I don't know enough about unicode, but https://github.com/jellyfin/jellyfin/issues/6393 requests `NormalizationForm.FormC`

**Comment by Jellifi007** [comment]
FormC algorithm is already applied to the string stripped from diacritics before returning it.

So in the end, new tests with Korean string from https://github.com/jellyfin/jellyfin/issues/6393 does not show any trace of aforementionned issue :
image

if that's ok, I will add those new tests with Korean strings

**Comment by Jellifi007** [comment]
About current AzurePipeline KO test status on all platforms : 
image
image

It looks like that #7624 merge broke some tests unrelated to this PR (see https://github.com/jellyfin/jellyfin/pull/7624/checks) :
image


(not sure if this is the right place to signal it)

File: src/Jellyfin.Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -7,6 +11,45 @@ namespace Jellyfin.Extensions
...
+            {
+                UnicodeCategory uc = CharUnicodeInfo.GetUnicodeCategory(c);
+                if (uc != UnicodeCategory.NonSpacingMark)
+                {
+                    withoutDiactritics.Append(c);
+                }
+            }
+
+            string normalizedText = withoutDiactritics.ToString().Normalize(NormalizationForm.FormC);
+            return normalizedText;
```
Comment (crobibero) [inline src/Jellyfin.Extensions/StringExtensions.cs:40]:
```suggestion
            return withoutDiactritics.ToString().Normalize(NormalizationForm.FormC);
```

**Review by crobibero — APPROVED** [review]

**Review by 1337joe — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**